overlays/haskell.nix

```diff
@@ -555,25 +555,30 @@ final: prev: {
                       });
                     }
                 ) rawProject.hsPkgs
-                // {
-                  # These are functions not packages
-                  inherit (rawProject.hsPkgs) shellFor makeConfigFiles ghcWithHoogle ghcWithPackages;
-              });
+                // (final.lib.mapAttrs (n:
+                      __trace "project.hsPkgs.${n} has been deprecated in favour of project.${n}"
```

Comment:
Can we also have a ChangeLog entry for this?

Reply:
Also, I thought from previous discussion we decided to not bother deprecating things and just remove them, so long as it was noted in the changelog. Admittedly, this one will break a *lot* of people, but not more than changing the top-level of `haskell.nix`, which was the last such thing we did.